instrumentation.go

```diff
@@ -293,3 +357,37 @@ func lookupServiceName() (string, bool) {
 
 	return "", false
 }
+
+// WithTraceExporter returns an [InstrumentationOption] that will configure an
+// [Instrumentation] to use the provided exp to export OpenTelemetry tracing
+// telemetry.
+//
+// If OTEL_TRACES_EXPORTER is defined, this option will conflict with
+// [WithEnv]. If both are used, the last one provided to an [Instrumentation]
+// will be used.
+func WithTraceExporter(exp trace.SpanExporter) InstrumentationOption {
+	return fnOpt(func(_ context.Context, c instConfig) (instConfig, error) {
+		c.traceExp = exp
+		return c, nil
+	})
+}
+
+func defaultTraceExporter(ctx context.Context) (trace.SpanExporter, error) {
+	// Controller-local reference to the auto-instrumentation release
+	// version. Start of this auto-instrumentation's exporter User-Agent
+	// header, e.g. ""OTel-Go-Auto-Instrumentation/1.2.3".
+	baseUserAgent := fmt.Sprintf("OTel-Go-Auto-Instrumentation/%s", Version())
+
+	// runtimeInfo is info about the runtime environment for inclusion in
+	// User-Agent, e.g. "go/1.18.2 (linux/amd64)".
+	v := strings.Replace(runtime.Version(), "go", "go/", 1)
+	runtimeInfo := fmt.Sprintf("%s (%s/%s)", v, runtime.GOOS, runtime.GOARCH)
```

Comment:
Can we remove this?

I think that this should come with the resource:
- https://opentelemetry.io/docs/specs/semconv/resource/host/
- https://opentelemetry.io/docs/specs/semconv/resource/process/

Also see: https://opentelemetry.io/docs/specs/otel/protocol/exporter/#user-agent